modules/@apostrophecms/schema/ui/apos/mixins/AposInputMixin.js

```diff
@@ -32,14 +44,24 @@ export default {
       error: false,
       // This is just meant to be sufficient to prevent unintended collisions
       // in the UI between id attributes
-      uid: Math.random()
+      uid: Math.random(),
+      // Automatically updated for you, can be watched
+      focus: false
     };
   },
   mounted () {
-    this.$el.addEventListener('focusout', this.validateAndEmit);
+    this.focusInListener = () => {
```

Comment:
Actually cannot be done that way because methods don't behave as methods when called as DOM event handlers (or from setTimeout, etc). And, I have to retain a reference to them in order to remove the event handlers later. So this appears to be the only way.

(Arrow functions use lexically scoped `this`, i.e. what it was inside `mounted`, which is what we want.)

Reply:
I would think it would be the same `this` in both cases.

Reply:
It is not because we'd be handing off `this.method` to `addEventListener`, but all it actually keeps is the `method` function, devoid of its context. The only way to make that work is to use `bind`, a closure, or an arrow function. All of these options work, but in all three cases we have to retain the bound function (even `bind` makes a new function) as a property for later cleanup. So, it doesn't matter which one we use.

I chose arrow functions because they are simple and this is an advertised feature of them (the language designers who put this feature in say: you should use arrow functions when you want a lexical `this`, it's one of the reasons we created it).

But I could use a `const self = this` pattern if you think that's better self-documentation. Makes no difference to me.

If there was more than one line of code in these functions I might be inclined to make them methods, and use `bind`, but I would still have to retain the bound function (which is a new function!) on `this` with an assignment just like you see here, because you have to in order to remove the listener later.

Reply:
I ran a test and they seems to work correctly when declared in `methods`. In both places `this` is the component. We do similar eventListener work in AposContextMenu, AposModal and AposTreeHeader.